### PR TITLE
Basic LLM connector.

### PR DESCRIPTION
**Pull request by h-alice**
Implemented a convenient llm connect, it supports Llamacpp and HuggingFace.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+langchain
```
Comment (NatLee) [inline requirements.txt:1]: Cool! First time to meet 🤯

File: src/core/llm_prompt_crafter.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+from langchain_core.documents.base import Document
+from langchain_core.language_models.llms import LLM
+
+from .common import LlmNames
+
+
+# Some prompt templates.
+PROMPT_FORMAT = {
+    LlmNames.LLAMA.value: {
+        "prompt": "<s>[INST] <<SYS>>\n{sys}\n<</SYS>>\n{full_user_prompt} [/INST]\n{model}\n",
```
Comment (NatLee) [inline src/core/llm_prompt_crafter.py:14]: This could be an Enum class instead of Dict. 🤩

**Review by NatLee — APPROVED** [review]
This is the first push!
Go! Go! Power ranger!

File: requirements.txt (hunk 1 above)
Comment (h-alice) [reply requirements.txt:1]: Langchain is convenient and good!
But there's still many bugs and poor documentation issues. : (

**Comment by h-alice** [comment]
> This is the first push! Go! Go! Power ranger!

Let's GOOOOOOO!

**Pull request merged**